optuna/study/_multi_objective.py

```diff
@@ -127,17 +126,18 @@ def _is_pareto_front_nd(unique_lexsorted_loss_values: np.ndarray) -> np.ndarray:
     loss_values = unique_lexsorted_loss_values[:, 1:]
     n_trials = loss_values.shape[0]
     on_front = np.zeros(n_trials, dtype=bool)
-    nondominated_indices: np.ndarray[tuple[int, ...], np.dtype[np.signedinteger[Any]]] = np.arange(
-        n_trials
-    )
```

Comment:
The same reason
- https://github.com/optuna/optuna/pull/5853#discussion_r1877207250